### PR TITLE
Primitive and ref

### PR DESCRIPTION
**Pull request by ValentinChCloud**
I open the PR and will fixed typo and sentences. Probably I will miss some faults, so don't hesitate to ping me here, or make PR.
Issue #15

### CODE REVIEW

File: pages/03-01 primitives-referenced-values.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+number
+boolean
+function
+nil
+string
+```
+
+### Composite data types
+
+In lua there is no composite data such as in other language as C/C++. But table in lua is a powerful type that can act as composite data type.
```
Comment (jaythomas) [inline pages/03-01 primitives-referenced-values.md]: Lua

File: pages/03-01 primitives-referenced-values.md [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    false
+
+    The strings equal and the numbers equal, but why aren't the tables and functions equal since they are both empty?
+```
+
+First, let's see variables types and their meanings.
+	
+	
+### Primitves data types
+In Lua, though we don't have variable data types, but we have types for the values. The list of data types for values are given below.
```
Comment (jaythomas) [inline pages/03-01 primitives-referenced-values.md]: Lua's data types is covered in the first chapter so listing the data types here won't be necessary. The focus should be on what the difference is between primitives and composites from a language-agnostic point of view.

File: pages/03-01 primitives-referenced-values.md [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+print(type("What is my type"))   --> string
+t = 10
+
+print(type(5.8*t))               --> number
+print(type(true))                --> boolean
+print(type(print))               --> function
+print(type(nil))                 --> nil
+print(type(type(ABC)))           --> string
+```
+When you build and execute the above program, it produces the following result on Linux −
```
Comment (jaythomas) [inline pages/03-01 primitives-referenced-values.md]: It should produce the same results on all operating systems, no?

**Comment by jaythomas** [comment]
Thanks for the pull request. I have left a fer comments above. Another thing I see is the "composite data type" section covers more about how to use a table as a composite data type without really defining what a composite data type is.
It's a little tricky to define as there is no strict definition, but I think the simplest explanation it can be boiled down to is that composite data is data composed of distinguishable parts, so a table that isn't empty contains primitives, functions, or even other composite tables.

**Comment by ValentinChCloud** [comment]
Thx for the comments, I will update.Yes the composite part was more complicated than I thought, it isn't as in C . I will clarify that

**Comment by jaythomas** [comment]
@ValentinChCloud this looks better. Were there any further changes you wished to make before this is merged in?

**Comment by ValentinChCloud** [comment]
No, you can merge it

**Pull request merged**